### PR TITLE
Created scroll to the top button

### LINKED ISSUE #190: Add scroll to top button
image

### PR DESCRIPTION
**Pull request by D1nviz**
image

https://github.com/idx-academy/spa-orders/assets/118522791/942d6cc4-a4a3-4dd0-9eff-e81814a4ec3f

### CODE REVIEW

File: code/src/components/scroll-to-top-button/ScrollToTopButton.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react";
+
+import { ArrowUpward } from "@mui/icons-material";
```
Comment (mhevyk) [inline code/src/components/scroll-to-top-button/ScrollToTopButton.tsx]: Use default import here

File: code/src/components/scroll-to-top-button/ScrollToTopButton.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+const ScrollToTopButton = () => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const toggleVisibility = () => {
+    if (window.scrollY >= SCROLL_THRESHOLD) {
+      setIsVisible(true);
+    } else {
+      setIsVisible(false);
+    }
```
Comment (mhevyk) [inline code/src/components/scroll-to-top-button/ScrollToTopButton.tsx]: We can simplify it with `setIsVisible(window.scrollY >= SCROLL_THRESHOLD)`

**Review by heshvab — APPROVED** [review]

File: code/src/components/scroll-to-top-button/ScrollToTopButton.tsx (hunk 1 above)
Comment (D1nviz) [reply code/src/components/scroll-to-top-button/ScrollToTopButton.tsx]: fixed

File: code/src/components/scroll-to-top-button/ScrollToTopButton.tsx (hunk 2 above)
Comment (D1nviz) [reply code/src/components/scroll-to-top-button/ScrollToTopButton.tsx]: fixed

**Review by OliaHrytsak — APPROVED** [review]

File: code/src/layouts/root-layout/RootLayout.tsx [hunk 3]
```diff
@@ -19,6 +20,7 @@ const RootLayout = () => {
           <Outlet />
         </Suspense>
       </AppBox>
+      <ScrollToTopButton />
```
Comment (mhevyk) [inline code/src/layouts/root-layout/RootLayout.tsx]: I suggest rendering it inside `App.tsx` next to `AppSnackbar` component if we want to have  to top button on every page regardless of layout it lives in

File: code/src/components/scroll-to-top-button/ScrollToTopButton.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    };
+
+    window.addEventListener("scroll", handleScroll);
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+  }, [setIsVisible]);
+
+  return (
+    <AppBox className={cn(`spa-scroll-button`, isVisible && "visible")}>
```
Comment (mhevyk) [inline code/src/components/scroll-to-top-button/ScrollToTopButton.tsx]: Is this `AppBox` necessary here? I noticed that if you click on button, the focus ripple is not covering all button. Is it expected behaviour?

File: code/src/layouts/root-layout/RootLayout.tsx (hunk 3 above)
Comment (D1nviz) [reply code/src/layouts/root-layout/RootLayout.tsx]: Moved

File: code/src/components/scroll-to-top-button/ScrollToTopButton.tsx (hunk 4 above)
Comment (D1nviz) [reply code/src/components/scroll-to-top-button/ScrollToTopButton.tsx]: fixed

**Review by mhevyk — APPROVED** [review]

**Pull request merged**